core/controllers/reader.py

```diff
@@ -1283,15 +1283,72 @@ class RecommendationsHandler(base.BaseHandler):
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
 
+    URL_PATH_ARGS_SCHEMAS = {
+        'exploration_id': {
+            'schema': {
+                'type': 'basestring',
+                'validators': [{
+                    'id': 'is_regex_matched',
+                    'regex_pattern': constants.ENTITY_ID_REGEX
+                }]
+            }
+        }
+    }
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {
+            'collection_id': {
+                'schema': {
+                    'type': 'basestring',
+                    'validators': [{
+                        'id': 'is_regex_matched',
+                        'regex_pattern': constants.ENTITY_ID_REGEX
+                    }]
+                },
+                'default_value': None
+            },
+            'include_system_recommendations': {
+                'schema': {
+                    'type': 'bool'
+                },
+                'default_value': True
+            },
+            'stringified_author_recommended_ids': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            },
+            'story_id': {
+                'schema': {
+                    'type': 'basestring',
+                    'validators': [{
+                        'id': 'is_regex_matched',
+                        'regex_pattern': constants.ENTITY_ID_REGEX
+                    }]
+                },
+                'default_value': None
```

Comment:
Why can this be `None`?

Reply:
Not sure what does that mean.

Reply:
@vojtechjelinek I changed my mind. I think `story_id` and `current_node_id` can also be `None` based on comments in the frontend file.
![Screen Shot 2022-06-21 at 10 02 00 PM](https://user-images.githubusercontent.com/31307813/174934450-01064ccb-3352-4c89-9206-9320d9eb642a.png)